packages/app/src/app/components/WorkspaceSetup/steps/SelectWorkspace.tsx

```diff
@@ -16,18 +16,35 @@ export const SelectWorkspace: React.FC<StepProps> = ({
   currentStep,
   numberOfSteps,
 }) => {
-  const { dashboard, activeTeam } = useAppState();
+  const { dashboard, activeTeam, user } = useAppState();
   const history = useHistory();
   const { setActiveTeam } = useActions();
   const { setQueryParam } = useURLSearchParams();
+  const { pathname } = useLocation();
+  const isSigningUpForBeta = pathname.includes('signup-beta');
 
-  const workspacesEligibleForUpgrade = dashboard.teams.filter(
-    t =>
-      !(
-        t.subscription?.status === SubscriptionStatus.Active ||
-        t.subscription?.status === SubscriptionStatus.Trialing
-      ) && t.featureFlags.ubbBeta
-  );
+  const workspacesEligibleForUpgrade = dashboard.teams
+    .filter(
```

Comment:
rewrote this filter for clarity. filters out workspaces that are not free and those for which you are not an admin, then it looks at the ubbFlag based on the `upgrade` vs `signup-beta` flow